spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2141,6 +2141,25 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
             scalarExprToProtoWithReturnType(algorithm, StringType, childExpr)
           }
 
+        case struct @ CreateNamedStruct(_) =>
+          val valExprs = struct.valExprs.map(exprToProto(_, inputs, binding))
+          val dataType = serializeDataType(struct.dataType)
+
+          if (valExprs.forall(_.isDefined) && dataType.isDefined) {
+            val structBuilder = ExprOuterClass.CreateNamedStruct.newBuilder()
+            structBuilder.addAllValues(valExprs.map(_.get).asJava)
+            structBuilder.setDatatype(dataType.get)
+
+            Some(
+              ExprOuterClass.Expr
+                .newBuilder()
+                .setCreateNamedStruct(structBuilder)
+                .build())
+          } else {
+            withInfo(expr, struct.valExprs: _*)
```

Comment:
I think we need to specify an info string here otherwise the user will not see the reason for falling back to Spark (perhaps @parthchandra can confirm).

```suggestion
            withInfo(expr, "unsupported arguments for CreateNamedStruct", struct.valExprs: _*)
```

Reply:
Based on this comment here https://github.com/apache/datafusion-comet/blob/7d6ad4bfe356e69751478ac5d0c4ad240966deb3/spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala#L1085-L1086 it sounds like we would still get the errors from the children which should contain the root issue. So it not clear to me we would be able to provide any additional useful information here.

Reply:
> I think we need to specify an info string here otherwise the user will not see the reason for falling back to Spark (perhaps @parthchandra can confirm).

The user will get the reason from the child nodes. We could add the additional message as @andygrove has suggested, but it is not strictly necessary.

Reply:
Resolved, by adding the message. I think there is some argument to made around this making it harder to figure out the root cause since there is more information in the info log.